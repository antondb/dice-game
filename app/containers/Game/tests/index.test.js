/**
 *
 * Tests for Game
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import { Provider } from 'react-redux';
import { Game } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';
// import 'jest-dom/extend-expect'; // add some helpful assertions
import configureStore from '../../../configureStore';

const props = {
  onRollDice: () => {},
  gameMeta: { gameMessage: 'a', gameState: 'b' },
  players: [],
};

describe('<Game />', () => {
  let store;

  beforeAll(() => {
    store = configureStore();
  });
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Game {...props} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });
  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Game {...props} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
