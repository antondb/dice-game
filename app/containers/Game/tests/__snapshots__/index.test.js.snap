// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> Should render and match the snapshot 1`] = `
<div
  class="Game__StyledGame-sc-1nrlr7e-1 khMrxA"
>
  <h1
    class="Title__StyledH1-sc-1q7q8nk-0 BJgVs"
  >
    Battle Dice
  </h1>
  <div
    class="Game__StyledPlayers-sc-1nrlr7e-0 cfgHeK"
  />
  <div
    class="Game__StyledButtonArea-sc-1nrlr7e-2 eQsbGk"
  >
    <p
      class="Game__StyledMessageArea-sc-1nrlr7e-3 gOLrMZ"
    >
      a
    </p>
    <span>
      <button
        class="Button__StyledInput-nl171n-0 liDtlR"
        type="button"
      >
        ROLL
      </button>
    </span>
  </div>
</div>
`;
